### PR TITLE
add abstract classes to the "class" test

### PR DESCRIPTION
**Pull request by evmar**
In preparation for #410, add some more test cases and comments to the "class" test.

No functionality change, just comments and renames.

### CODE REVIEW

File: test_files/class/class.ts [hunk 1]
```diff
@@ -1,24 +1,52 @@
...
+// This tests "implements" against an abstract class.
+// Note: in Closure you cannot currently "implements" two classes, see #410.
+class ImplementsAbstract implements AbstractClass {
+  abstractFunc(): void {}
+  // Note: because this class *implements* AbstractClass, it must also implement
+  // nonAbstractFunc despite that already having an implementation.
+  nonAbstractFunc(): void {}
+}
+
+// This tests "extends" against the class.
```
Comment (mprobst) [inline test_files/class/class.ts]: ... and implements?

File: test_files/class/class.ts [hunk 2]
```diff
@@ -1,24 +1,52 @@
...
+class Extends extends Class implements Interface {
   interfaceFunc(): void {}
 }
 
+// This tests "extends" against the abstract class.
+class ExtendsAbstract extends AbstractClass {
```
Comment (mprobst) [inline test_files/class/class.ts]: should these be `ClassExtendsAbstract`, so we can also have `InterfaceExtendsInterface` vs `InterfaceImplementsInterface`? That is, put the Cartesian product into the type names to describe the case it's exercising?

Comment (evmar) [inline test_files/class/class.ts]: I didn't realize you could "extends" an interface!  I will fix.

**Comment by evmar** [comment]
PTAL (just review the .ts test)

File: test_files/class/class.ts [hunk 3]
```diff
@@ -1,38 +1,107 @@
...
+// 1) interface implements
+// 2) interface extends
+// 3) class implements
+// 4) class extends
 
-class Implements implements Interface, Super {
-  interfaceFunc(): void {}
-  superFunc(): void {}
+// 1) interface implements.
+// No examples; this is not legal TypeScript.
```
Comment (mprobst) [inline test_files/class/class.ts]: I don't understand what this comment refers to. Should this have a number from above?

File: test_files/class/class.ts [hunk 4]
```diff
@@ -1,38 +1,107 @@
...
+abstract class AbstractClass {
+  abstract abstractFunc(): void;
+  nonAbstractFunc(): void { }
 }
 
+// Write out all permutations:
+// 1) interface implements
+// 2) interface extends
+// 3) class implements
+// 4) class extends
```
Comment (mprobst) [inline test_files/class/class.ts]: does it make any difference whether the class extending/implementing is abstract? If not, document?

File: test_files/class/class.ts (hunk 3 above)
Comment (evmar) [inline test_files/class/class.ts]: Maybe it's not worth numbering these.  I will remove.

File: test_files/class/class.ts (hunk 4 above)
Comment (evmar) [inline test_files/class/class.ts]: It probably can matter.  I'll add some more sections.

**Comment by evmar** [comment]
PTAL

**Pull request merged**